demo/api/src/importer/data-streams/data-stream.ts

```diff
@@ -1,9 +1,11 @@
-export interface DataStreamAndMetadata {
-    dataStream: NodeJS.ReadableStream;
-    params?: Record<string, unknown>;
-    additionalData?: Record<string, unknown>;
-}
+import { Readable } from "stream";
+
+import { PipeMetadata } from "../pipes/importer-pipe.type";
 
+export type StreamChunkAndMetadata = {
+    chunk: Buffer | string;
+    metadata: PipeMetadata<Record<string, unknown>>;
+};
 export abstract class DataStream {
-    abstract getDataStreamAndMetadata(): Promise<DataStreamAndMetadata | null>;
+    abstract getDataStreamAndMetadata(): Promise<Readable | null>;
```

Comment:
Can we use a more specific type than Readable? Doesn't it have to be the shape of StreamChunkAndMetadata?